spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -38,6 +39,12 @@ case object MklDnn extends EngineType
 
 
 object Engine {
+
+  // initialize some properties. Do not call it by manually.
+  // Because when we create some variables, they will set relevant properties, such as `_model`.
+  // If we do, any change of property for mkldnn will have no effect.
+  setMklDnnEnvironments()
```

Comment:
It looks a little confused. what about if(System.property("...") == mkldnn) setMklDnnEnvironments?